dotnet/src/IntegrationTests/Connectors/Memory/Qdrant/QdrantVectorStoreFixture.cs

```diff
@@ -0,0 +1,325 @@
+        List<PointStruct> unnamedVectorPoints =
+        [
+            new PointStruct
+            {
+                Id = 11,
+                Vectors = embedding,
+                Payload = { ["HotelName"] = "My Hotel 11", ["HotelCode"] = 11, ["ParkingIncluded"] = true, ["Tags"] = tagsValue, ["HotelRating"] = 4.5f, ["Description"] = "This is a great hotel." }
+            },
+            new PointStruct
+            {
+                Id = 12,
+                Vectors = embedding,
+                Payload = { ["HotelName"] = "My Hotel 12", ["HotelCode"] = 12, ["ParkingIncluded"] = false, ["Description"] = "This is a great hotel." }
+            },
+            new PointStruct
+            {
+                Id = 13,
+                Vectors = embedding,
+                Payload = { ["HotelName"] = "My Hotel 13", ["HotelCode"] = 13, ["ParkingIncluded"] = false, ["Description"] = "This is a great hotel." }
+            },
+        ];
+
+        await this.QdrantClient.UpsertAsync("singleVectorHotels", unnamedVectorPoints);
+
+        // Create some test data using a single unnamed vector and a guid id.
+        List<PointStruct> unnamedVectorGuidIdPoints =
+        [
+            new PointStruct
+            {
+                Id = Guid.Parse("11111111-1111-1111-1111-111111111111"),
+                Vectors = embedding,
+                Payload = { ["HotelName"] = "My Hotel 11", ["Description"] = "This is a great hotel." }
+            },
+            new PointStruct
+            {
+                Id = Guid.Parse("22222222-2222-2222-2222-222222222222"),
+                Vectors = embedding,
+                Payload = { ["HotelName"] = "My Hotel 12", ["Description"] = "This is a great hotel." }
+            },
+            new PointStruct
+            {
+                Id = Guid.Parse("33333333-3333-3333-3333-333333333333"),
+                Vectors = embedding,
+                Payload = { ["HotelName"] = "My Hotel 13", ["Description"] = "This is a great hotel." }
+            },
+        ];
+
+        await this.QdrantClient.UpsertAsync("singleVectorGuidIdHotels", unnamedVectorGuidIdPoints);
+    }
+
+    /// <summary>
+    /// Delete the docker container after the test run.
+    /// </summary>
+    /// <returns>An async task.</returns>
+    public async Task DisposeAsync()
+    {
+        if (this._containerId != null)
+        {
+            await this._client.Containers.StopContainerAsync(this._containerId, new ContainerStopParameters());
+            await this._client.Containers.RemoveContainerAsync(this._containerId, new ContainerRemoveParameters());
+        }
+    }
+
+    /// <summary>
+    /// Setup the qdrant container by pulling the image and running it.
+    /// </summary>
+    /// <param name="client">The docker client to create the container with.</param>
+    /// <returns>The id of the container.</returns>
+    private static async Task<string> SetupQdrantContainerAsync(DockerClient client)
```

Comment:
Not related to this PR, but for integration tests in general - with increased number of integration tests, our merge pipeline job will take more time, and sometimes it will block PR from merging because of possible failed tests. It's a good approach, if failed test is related to changes in new PR, but in our case, if latest `qdrant/qdrant` or any other vector db image will include breaking changes, it will block all our PRs from merging until the problem will be fixed.

I think we should perform following updates to our pipelines:
1. We need to run integration tests that are related to changes in PR only. We can check file path to understand which tests to run.
2. If we introduce previous point, there will be another problem, when we don't touch some file/functionality for long period of time, but we still need to run tests from time to time. For these cases, we should introduce background jobs that will run all tests regularly to verify that all scenarios work as expected. 

This should be done probably before merging `feature-memory-service` branch to `main`, in order to avoid a situation described above. Or, we can keep all vector db integration tests as disabled, for manual verification only, merge `feature-memory-service` branch to `main` and after that update infrastructure for integration tests and enable them again.

Reply:
I agree with these @dmytrostruk, both 1 & 2 make a lot of sense.  We should also consider running against specific versions, instead of the latest to avoid being broken when a breaking change is released by a db vendor.  I'll add it to the list.